### DISCUSSION TITLE
Is it planned to integrate an approach to use generic REST API in Automations/as Data Sources?

### DISCUSSION THREAD

**Discussion by tlindener** [nocodb/nocodb#4859] [Features]
I'm wondering if there are plans to genrally integrate REST APIs as part of Automations and data sources?

**Comment by o1lab** [comment]
Not at the moment. However could you please expand on REST API as data sources with you usecase ?

**Reply by tlindener** [reply]
Hey thanks for the quick reply!

I have a couple of uses cases in my mind right now where I would like to use a no code/low code platform:

1. I have a table of entities in another system now, I would like the user to be able to do an approval either of a recommendation generated from another system
2. I have a table of entities, I want the user to invoke an API call and get additional data related to that entity. If the data are useful, the user should be able to save this into a related table
3. The user should be able to maintain a dataset where I want to generate events to a Kafka system (e.g. via API)
4. I retrieve a table via an API and the user should select rows/data which should be imported into another system

Is this within the scope of the project?

**Reply by tlindener** [reply]
@o1lab Hey,
Just to follow up on this. Is this something that might be part of nocodb in the future? May be something that can be added as an OS contribution?
